zedd-app/src/components/AppBody.tsx

```diff
@@ -247,7 +251,22 @@ export const AppBody = observer(
             </Button>
             <ButtonGroup variant='outlined'>
               <Tooltip title='Fill currently shown empty days with ERSATZ task.' arrow>
-                <Button size='large' onClick={(_) => state.fillErsatz(state.showing)}>
+                <Button
+                  size='large'
+                  onClick={(_) => {
+                    if (settings.location) {
+                      getHolidays(
+                        (state.showing.start as Date).getFullYear(),
```

Comment:
you can use date-fns' getYear to avoid casting

Reply:
Additionally, this doesn't handle when state.showing spans multiple years. I would suggest an additional function getRangeHolidays, which uses eachYearOfInterval and getHolidays.